File: src/main/java/com/example/debug/engine/model/Configuration.java
package com.example.debug.engine.model;

import com.fasterxml.jackson.annotation.JsonIgnore;

/* loaded from: classes.dex */
public class Configuration {
    private boolean considerOrientation;
    private int cutOrientationPreference;
    private String cutThickness;
    private String minTrimDimension;
    private double optimizationFactor;
    private int optimizationPriority;

    @JsonIgnore
    private PerformanceThresholds performanceThresholds;
    private Integer units;
    private boolean useSingleStockUnit;

    public String getCutThickness() {
        return this.cutThickness;
    }

    public void setCutThickness(String str) {
        this.cutThickness = str;
    }

    public boolean isUseSingleStockUnit() {
        return this.useSingleStockUnit;
    }

    public void setUseSingleStockUnit(boolean z) {
        this.useSingleStockUnit = z;
    }

    public double getOptimizationFactor() {
        return this.optimizationFactor;
    }

    public void setOptimizationFactor(double d) {
        this.optimizationFactor = d;
    }

    public int getOptimizationPriority() {
        return this.optimizationPriority;
    }

    public void setOptimizationPriority(int i) {
        this.optimizationPriority = i;
    }

    public int getCutOrientationPreference() {
        return this.cutOrientationPreference;
    }

    public void setCutOrientationPreference(int i) {
        this.cutOrientationPreference = i;
    }

    public Integer getUnits() {
        return this.units;
    }

    public void setUnits(Integer num) {
        this.units = num;
    }

    public boolean isConsiderOrientation() {
        return this.considerOrientation;
    }

    public void setConsiderOrientation(boolean z) {
        this.considerOrientation = z;
    }

    public String getMinTrimDimension() {
        return this.minTrimDimension;
    }

    public void setMinTrimDimension(String str) {
        this.minTrimDimension = str;
    }

    public PerformanceThresholds getPerformanceThresholds() {
        return this.performanceThresholds;
    }

    public void setPerformanceThresholds(PerformanceThresholds performanceThresholds) {
        this.performanceThresholds = performanceThresholds;
    }

    public String toString() {
        return "Configuration{cutThickness=" + this.cutThickness + ", useSingleStockUnit=" + this.useSingleStockUnit + ", optimizationFactor=" + this.optimizationFactor + ", optimizationPriority=" + this.optimizationPriority + ", units=" + this.units + ", considerOrientation=" + this.considerOrientation + '}';
    }
}

File: src/features/engine/model/configuration.rs
use serde::{Deserialize, Serialize};

use crate::{enums::{cut_orientation_preference::CutOrientationPreference, optimization_level::OptimizationFactor, optimization_priority::OptimizationPriority, orientation::Orientation}, features::engine::model::performance_thresholds::PerformanceThresholds};


#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Configuration {
    pub consider_orientation: bool,
    pub cut_orientation_preference: CutOrientationPreference,
    pub cut_thickness: Option<String>,
    pub min_trim_dimension: Option<String>,
    pub optimization_factor: OptimizationFactor,
    pub optimization_priority: OptimizationPriority,
    
    #[serde(skip_serializing_if = "Option::is_none")]
    pub performance_thresholds: Option<PerformanceThresholds>,
    
    pub units: Option<i32>,
    pub use_single_stock_unit: bool,
}

impl Default for Configuration {
    fn default() -> Self {
        Self {
            consider_orientation: false,
            cut_orientation_preference: CutOrientationPreference::default(),
            cut_thickness: None,
            min_trim_dimension: None,
            optimization_factor: OptimizationFactor::default(),
            optimization_priority: OptimizationPriority::default(),
            performance_thresholds: None,
            units: None,
            use_single_stock_unit: false,
        }
    }
}
